File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/genres/model/GenreList.java
package dev.reuss.tmdb.domain.genres.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.genre.Genre;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * List of TMDB genres.
 *
 * @param genres genres
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record GenreList(
        List<Genre> genres
) implements TmdbModel {

    public GenreList {
        genres = TmdbCollections.list(genres);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/genres/model/GenreList.kt
package dev.reuss.tmdb.domain.genres.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.genre.Genre

/**
 * List of TMDB genres.
 *
 * @property genres genres
 */
data class GenreList(
    @all:JsonProperty("genres")
    val genres: List<Genre>?
) : TmdbModel